examples/multigpu_advtrain/target_attack.py

```diff
@@ -0,0 +1,257 @@
+import numpy as np
+import tensorflow as tf
+
+from cleverhans.model import Model, CallableModelWrapper
+from cleverhans.attacks import Attack, FastGradientMethod
+from cleverhans.utils_tf import model_loss
+
+from model import clone_variable, MLP_probs
+
+
+class ProjectedGradientDescentMethod(Attack):
+
+    """
+    The Basic Iterative Method (Kurakin et al. 2016). The original paper used
+    hard labels for this attack; no label smoothing.
+    Paper link: https://arxiv.org/pdf/1607.02533.pdf
+    """
+
+    def __init__(self, model, back='tf', sess=None):
+        """
+        Create a ProjectedGradientDescentMethod instance.
+        """
+        super(ProjectedGradientDescentMethod, self).__init__(model, back, sess)
+        self.feedable_kwargs = {'eps': np.float32,
+                                'eps_iter': np.float32,
+                                'y': np.float32,
+                                'y_target': np.float32,
+                                'clip_min': np.float32,
+                                'clip_max': np.float32}
+        self.structural_kwargs = ['ord', 'nb_iter']
+
+        if not isinstance(self.model, Model):
+            self.model = MLP_probs(self.model)
+
+    def clip_eta(self, eta):
+        # Clipping perturbation eta to self.ord norm ball
+        if self.ord == np.inf:
+            eta = tf.clip_by_value(eta, -self.eps, self.eps)
+        elif self.ord in [1, 2]:
+            reduc_ind = list(xrange(1, len(eta.get_shape())))
+            if self.ord == 1:
+                norm = tf.reduce_sum(tf.abs(eta),
+                                     reduction_indices=reduc_ind,
+                                     keep_dims=True)
+            elif self.ord == 2:
+                norm = tf.sqrt(tf.reduce_sum(tf.square(eta),
+                                             reduction_indices=reduc_ind,
+                                             keep_dims=True))
+            eta = eta * self.eps / norm
+        return eta
+
+    def PGDpgd(self, x, **kwargs):
```

Comment:
What does this method name mean?

Reply:
Fixed in 460d3f3. Changed to `attack` to make similar to the code in attack.py.